### PR TITLE
refac: Start calculation when batch created

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

Resolve these problems in `BatchApplicationService.cs` by introducing batch-created-events:

image

## References

* #199

### CODE REVIEW

File: source/dotnet/Services/IntegrationTests/ProcessManager/BatchApplicationServiceTests.cs [hunk 1]
```diff
@@ -57,6 +57,7 @@ public BatchApplicationServiceTests(ProcessManagerDatabaseFixture processManager
         _databricksWheelClientMock.Setup(x => x.Jobs).Returns(_jobsApiMock.Object);
     }
 
+    /* TODO AJW
```
Comment (lasrinnil) [inline source/dotnet/Services/IntegrationTests/ProcessManager/BatchApplicationServiceTests.cs:60]: TODO

File: source/dotnet/Services/IntegrationTests/WebApi/BatchControllerTests.cs [hunk 2]
```diff
@@ -54,14 +57,26 @@ public Task DisposeAsync()
         return Task.CompletedTask;
     }
 
+    /* TODO AJW
```
Comment (lasrinnil) [inline source/dotnet/Services/IntegrationTests/WebApi/BatchControllerTests.cs:60]: TODO

File: source/dotnet/Services/ProcessManager/Program.cs [hunk 3]
```diff
@@ -130,22 +130,23 @@ private static void Infrastructure(IServiceCollection serviceCollection)
             }));
 
         var serviceBusConnectionString =
-            EnvironmentVariableHelper.GetEnvVariable(EnvironmentSettingNames.ServiceBusSendConnectionString);
+            EnvironmentVariableHelper.GetEnvVariable(EnvironmentSettingNames.ServiceBusManageConnectionString);
+        var messageTypes = new Dictionary<Type, string>
+        {
+            //{ typeof(BatchCreatedDomainEventDto), EnvironmentVariableHelper.GetEnvVariable(EnvironmentSettingNames.BatchCreatedEventName) },
```
Comment (lasrinnil) [inline source/dotnet/Services/ProcessManager/Program.cs:136]: commented

File: source/dotnet/Services/WebApi/Configuration/ServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -101,6 +103,20 @@ public static void AddCommandStack(this IServiceCollection services, IConfigurat
...
         services.AddScoped<IJsonNewlineSerializer, JsonNewlineSerializer>();
+        services.AddScoped<ICorrelationContext, CorrelationContext>();
+        services.AddScoped<IJsonSerializer, JsonSerializer>();
+
+        var serviceBusConnectionString =
+            EnvironmentVariableHelper.GetEnvVariable(EnvironmentSettingNames.ServiceBusSendConnectionString);
+        var messageTypes = new Dictionary<Type, string>
+        {
+            { typeof(BatchCreatedDomainEventDto), EnvironmentVariableHelper.GetEnvVariable(EnvironmentSettingNames.BatchCreatedEventName) },
+            //{ typeof(BatchCompletedEventDto), EnvironmentVariableHelper.GetEnvVariable(EnvironmentSettingNames.BatchCompletedEventName) },
```
Comment (lasrinnil) [inline source/dotnet/Services/WebApi/Configuration/ServiceCollectionExtensions.cs:114]: commented

File: source/dotnet/Services/WebApi/EnvironmentSettingNames.cs [hunk 5]
```diff
@@ -35,5 +35,26 @@ public static class EnvironmentSettingNames
         public const string CalculationStorageContainerName = "STORAGE_CONTAINER_NAME";
 
         public const string DateTimeZoneId = "TIME_ZONE";
+
+        #region ServiceBus
+
+        // TODO: Add health check
```
Comment (lasrinnil) [inline source/dotnet/Services/WebApi/EnvironmentSettingNames.cs:41]: TODO

File: source/dotnet/Services/WebApi/EnvironmentSettingNames.cs [hunk 6]
```diff
@@ -35,5 +35,26 @@ public static class EnvironmentSettingNames
...
+        // TODO: Add these new settings in terraform
```
Comment (lasrinnil) [inline source/dotnet/Services/WebApi/EnvironmentSettingNames.cs:42]: TODO (important)

File: source/dotnet/Services/WebApi/Startup.cs [hunk 7]
```diff
@@ -49,7 +51,7 @@ public void ConfigureServices(IServiceCollection services)
         services.AddJwtTokenSecurity();
         services.AddCommandStack(Configuration);
         services.AddApplicationInsightsTelemetry();
-
+        ReRegisterCorrelationIdContextHack(services);
```
Comment (lasrinnil) [inline source/dotnet/Services/WebApi/Startup.cs:54]: What should we call this little maneuver?

File: source/dotnet/Services/Infrastructure/EventPublishers/ServiceCollectionExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+    public static IServiceCollection AddIntegrationEventPublisher(
+        this IServiceCollection serviceCollection,
+        string serviceBusConnectionString,
+        string topicName,
+        IDictionary<Type, string> messageType)
+    {
+        serviceCollection.AddScoped<IProcessCompletedIntegrationEventPublisher, ProcessCompletedIntegrationEventPublisher>();
+
+        if (serviceCollection.All(x => x.ServiceType != typeof(ServiceBusClient)))
```
[the remark this answers is not in the document]
Comment (lasrinnil) [reply source/dotnet/Services/Infrastructure/EventPublishers/ServiceCollectionExtensions.cs:68]: done

File: source/dotnet/Services/ProcessManager/Program.cs (hunk 3 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/ProcessManager/Program.cs]: Fixed by removing commented code.
The difference between the registration in web API and process manager is slightly non-intuitive.

File: source/dotnet/Services/WebApi/Configuration/ServiceCollectionExtensions.cs (hunk 4 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/WebApi/Configuration/ServiceCollectionExtensions.cs]: Fixed by removing commented code.
The difference between the registration in web API and process manager is slightly non-intuitive.

File: source/dotnet/Services/IntegrationTests/ProcessManager/BatchApplicationServiceTests.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/IntegrationTests/ProcessManager/BatchApplicationServiceTests.cs]: Changed to skipped.

File: source/dotnet/Services/IntegrationTests/WebApi/BatchControllerTests.cs (hunk 2 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/IntegrationTests/WebApi/BatchControllerTests.cs]: Changed to skipped.

File: source/dotnet/Services/WebApi/EnvironmentSettingNames.cs (hunk 5 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/WebApi/EnvironmentSettingNames.cs]: Health checks added.

File: source/dotnet/Services/WebApi/EnvironmentSettingNames.cs (hunk 6 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/WebApi/EnvironmentSettingNames.cs]: Added by you 😄

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**